gigascience/gigadb-website
# Enable display of github links on dataset page and add a test scenario

# Pull request for issue: #855

This is a pull request for the following functionalities:

Github links not displayed on dataset page under the external links section


## How to test?

Navigate to ``http://gigadb.gigasciencejournal.com:9170/dataset/100935``

You should see a section for Github links and a link to a GitHub repo below	

## How have functionalities been implemented?

In ``protected/views/dataset/view.php``

I have added the "Github Links" external type name to the list of external link types to display as parameter to the ``getDatasetExternalLinksTypesAndCount()`` method.

## Any issues with implementation?

none

## Any changes to automated tests?

Added an acceptance test scenario for the implemented functionality in ``tests/acceptance/DatasetView.feature``.
Also fixed an erroneous link in the project image scenario (it's correct on develop already , but there was a mishap during a past rebase) 

Added a new docker compose service for modernised Selenium/WebDriver functionality to work with Apple silicon based Mac. 
Added the same specific IP address to the ``chrome`` and ``chrome-arm`` docker service  and configure the acceptance tests to use that IP address irrespective of which of the two service is actually launched by ``up.sh``.
up.sh is the one that will detect the architecture and choose one over the other based on that.

All tests runners should pass whether we are running on Intel or Apple Silicon based Mac.
Relevant changes are in:

* ops/deployment/docker-compose.yml
* up.sh
* tests/acceptance_runner
* tests/acceptance.suite.yml
* behat.yml

To test that, executing all runners should work as usual:
```
$ ./tests/unit_runner && ./tests/functional_runner && ./tests/unit_functional_runner
$ ./tests/acceptance_runner local 
```
